src/visualizers/NumberGlyph.ts

```diff
@@ -160,6 +160,9 @@ class NumberGlyph extends P5Visualizer(paramDesc) {
         try {
             parsetree = math.parse(params.formula)
         } catch (err: unknown) {
+            // TODO in overhaul: make a utility somewhere that both this
+            // SequenceFormula.ts can use; also move this into the individual
+            // param validator for the 'formula' param
```

Comment:
"both this and"?

Reply:
Well, given that you've also made a very reasonable request to adjust the Formula validation, rather than make that change in both places I am actually going to go ahead and do this "TODO". There are two ways to go about it.

1. Add a `ParamType.FORMULA` in src/shared/ParamType.ts, or
2. Add a convenience function `validateFormula` in src/sequences/Formula.ts that can be separately imported other places that want to have `STRING` parameters that really should be formulas.

Thoughts/preferences on which way to go? I guess it depends on how common we expect it will be that people will want formula parameters and whether the additional ParamType will be more complicated/confusing/entry-barrier-creating than useful.

Reply:
Hmmmm, I think a ParamType.FORMULA seems useful (we've already got one place we need this).  I could imagine using it for a colour range on Turtle path for example.... and for the shiftcompare, you could have any formula for whether to turn a pixel on... I can think of various uses!

Reply:
OK, working on it as a ParamType.

Reply:
OK, now there is a ParamType.FORMULA. Note that if the inputs are anything but the usual `n`, they must be specified by an `inputs` property; see NumberGlyph `growthFormula` parameter for an example.

Reply:
Nice!